internal/graph/dotgraph_test.go

```diff
@@ -138,6 +138,22 @@ func TestComposeWithStandardGraphAndURL(t *testing.T) {
 	compareGraphs(t, buf.Bytes(), "compose6.dot")
 }
 
+func TestComposeWithNamesThatNeedEscaping(t *testing.T) {
+	g := baseGraph()
+	a, c := baseAttrsAndConfig()
+
+	g.Nodes[0].Info = NodeInfo{Name: "var\"src\""}
+	g.Nodes[1].Info = NodeInfo{Name: "var\"#dest#\""}
+
+	// Set edge to be Residual.
```

Comment:
Comment on the "why", not about the "what". It's clear that the line below sets the residual to true, it's unclear why.

Reply:
Thanks! I'm not sure "why" either, unfortunately! 😭 

I copied this test verbatim from here, and then adapted it for what i wanted to test:
https://github.com/google/pprof/blob/dff1afcd898c3bc6799f02ca45da17ba49dea69c/internal/graph/dotgraph_test.go#L67-L93

Do you know why they set the edge to be Residual? do you know what it means and/or what it does? 😬 

I think i tried deleting it but the test failed, but i don't remember why.

Reply:
No, I don't know why off top of my head. What was the failure without it?

Reply:
Hmm, i dunno! It just changes what the edge prints like:
```diff
        -N1 -> N2 [label=" 10" weight=11 color="#b28559" tooltip="var\"src\" -> var\"#dest#\" (10)" labeltooltip="var\"src\" -> var\"#dest#\" (10)"]
        +N1 -> N2 [label=" 10" weight=11 color="#b28559" tooltip="var\"src\" ... var\"#dest#\" (10)" labeltooltip="var\"src\" ... var\"#dest#\" (10)" style="dotted"]
```

But hmm, yeah, i don't think this is important to our test at all, so maybe i'll just delete it, and change the expected graph to match whatever it's printing? :)

Or i can keep it and add a comment along the lines of `// We mark this edge residual to follow the same behavior as the other tests. We don't know if it's needed, but this test is only focused on the string escaping.` 😅 

I'm supportive of deleting the line and changing the expected result if that works for you!

Reply:
Unfortunately, I won't have time to figure this out but I'm also a bit hesitant to either put partial or "we do this but we don't know why" comment. So you can either wait till I have time to look at it or you need to dive a bit more into this and figure this out. It is a bit unfair to ask you to block on this since you are using a copy of a pre-existing comment, but I prefer to use every PR to improve the code a tiny bit. Sorry about this!

Reply:
Done. Fixed in 060ba7359877a1faf326f1637ab8f802ecb95f6d.

It turns out that this was just because I was copying from a test that was _specifically testing_ that Residual edges work as expected. This was unnecessary info for my test, so i've deleted that line and adjusted the expected output accordingly. 👍 Thanks! :)